libsplinter/src/circuit/template/rules/create_services.rs

```diff
@@ -50,3 +132,187 @@ impl From<v1::ServiceArgument> for ServiceArgument {
         }
     }
 }
+
+fn add1_char(c: char) -> Result<String, CircuitTemplateError> {
+    let char_value = c as u32;
+    if char_value >= 'z' as u32 {
+        return Ok('0'.to_string());
+    }
+    let next_char = std::char::from_u32(char_value + 1)
+        .ok_or_else(|| CircuitTemplateError::new("Failed to generate service id"))?;
```

Comment:
This error message should probably be more specific; if it occurs, it should be clear where in the code it came from. I think this error is impossible to reach, but just in case.

Reply:
Addressed